daemon/main.py

```diff
@@ -0,0 +1,50 @@
+import argparse
+import asyncio
+import json
+from aiokafka import AIOKafkaProducer
+
+from fetchers import FetcherFactory
+from periodic import PeriodicRunner
+from config import Config
+
+
+async def main(
+    config_path: str, kafka_servers: str, kafka_topic: str, max_queue_size: int = 0
+):
+    cfg = Config.from_file(config_path)
+    queue = asyncio.Queue(maxsize=max_queue_size)
+    runners = []
+
+    for fcfg in cfg:
+        fetcher = FetcherFactory(fcfg["type"], **fcfg["args"])
+        runner = PeriodicRunner(fetcher, queue, fcfg["period"])
+        runners.append(runner)
+        runner.start()
+
+    producer = AIOKafkaProducer(
+        bootstrap_servers=kafka_servers,
+        loop=asyncio.get_running_loop(),
```

Comment:
Please do not use AIOKafka. 
We have a standardized Kafka stack (Arroyo) that has its own producer that manages callbacks and queue for you and runs asynchronously. It does not support AIO though so the two would not play well together as the AIOProducer would be used synchronously thereby dropping all the gains.